src/ORM/DataList.php

```diff
@@ -1964,7 +1962,7 @@ public function getIDList()
      */
     public function relation($relationName)
     {
-        $ids = $this->column('ID');
+        $ids = $this->sort(null)->column('ID');
```

Comment:
```suggestion
        $ids = $this->column('ID');
```
Sort order needs to be respected here

e.g.

When I have a `MyDataObject` that hasMany `MySubDataObjects`

And `MySubDataObjects` has a `default_sort` of `Title DESC`

When I go `MyDataObject::get()->relation('MySubDataObjects')->column()`

Then I expect the IDs to be sorted by "Title DESC"

Reply:
Reverted, will add a unit test to protect against regressions.